src/models/leafletButtonControl.ts

```diff
@@ -20,11 +31,27 @@ export class ButtonControl extends Control {
     button.innerHTML = `<i class='bi ${this.iconKey}'></i>`
     DomEvent.disableClickPropagation(button)
     DomEvent.on(button, 'click', () => {
-      this.state.value = button.classList.toggle(this.activeClass)
+        this.state.value = !this.state.value
     })
 
     container.title = this.tooltipTitle
 
+    watch(this.state, () => {
+      let changeType: StateChangeType
+
+      if (this.state.value) {
+        button.classList.add(this.activeClass)
+        changeType = StateChangeType.ACTIVATION
+      } else {
+        button.classList.remove(this.activeClass)
+        changeType = StateChangeType.DEACTIVATION
+      }
+
+      if (this.stateChangeHook) {
+        this.stateChangeHook(this.state, changeType)
+      }
+    })
```

Comment:
IMO a watcher is to much overhead here. Everytime the onAdd is called a new watcher is generated. So it is possible to create multiple watcher for the same state ref. I would move the logic to the place where the state is changed (line 34).

Reply:
The watcher is needed to ensure that `state` and the `active` HTML-class are synchronized. If the logic is simply moved to the state change, the HTML-class will only be updated when the control is clicked and not when the state is changed from the outside. The watcher creation should be moved to the constructor though, so we don't have multiple watchers for one ref.